framework/application/display_window.cpp

```diff
@@ -0,0 +1,285 @@
+/*
+** Copyright (c) 2021 Broadcom, Inc.
+** Copyright (c) 2021 LunarG, Inc.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and associated documentation files (the "Software"),
+** to deal in the Software without restriction, including without limitation
+** the rights to use, copy, modify, merge, publish, distribute, sublicense,
+** and/or sell copies of the Software, and to permit persons to whom the
+** Software is furnished to do so, subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be included in
+** all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#include "application/display_window.h"
+#include "application/application.h"
+
+#include "util/logging.h"
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(application)
+
+DisplayWindow::DisplayWindow(DisplayContext* display_context) : display_context_(display_context)
+{
+    assert(display_context_ != nullptr);
+}
+
+VkResult DisplayWindow::SelectPhysicalDevice(const encode::InstanceTable* table,
+                                             VkInstance                   instance,
+                                             VkPhysicalDevice*            physical_device) const
+{
+    assert(table);
+    uint32_t physical_device_count = 0;
+    VkResult error                 = table->EnumeratePhysicalDevices(instance, &physical_device_count, nullptr);
+    if (error || !physical_device_count)
```

Comment:
This test and the body are somewhat difficult for me to read.  I think they combine reporting of two conditions in the name of brevity, but I prefer it to be clearer.  Can this and other similar clauses be factored out into something like this?
```
if (error != VK_SUCCESS) {
    return error;
}
if(physical_device_count == 0)
{
    return VK_ERROR_INITIALIZATION_FAILED;
}
```

Reply:
Done.

Reply:
👍